tornado/httputil.py

```diff
@@ -230,13 +239,127 @@ def parse(cls, headers):
+        self._dict.clear()
+        self._as_list.clear()
+        self._last_key = None
+
+    def update(*args, **kwds):
+        ''' D.update([E, ]**F) -> None.  Update D from mapping/iterable E and F.
+            If E present and has a .keys() method, does:     for k in E: D[k] = E[k]
+            If E present and lacks .keys() method, does:     for (k, v) in E: D[k] = v
+            In either case, this is followed by: for k, v in F.items(): D[k] = v
+        '''
+        if len(args) > 2:
+            raise TypeError("update() takes at most 2 positional "
+                            "arguments ({} given)".format(len(args)))
+        elif not args:
+            raise TypeError("update() takes at least 1 argument (0 given)")
+        self = args[0]
+        other = args[1] if len(args) >= 2 else ()
+
+        if isinstance(other, collections.Mapping):
+            for key in other:
+                self[key] = other[key]
+        elif hasattr(other, "keys"):
+            for key in other.keys():
+                self[key] = other[key]
+        else:
+            for key, value in other:
+                self[key] = value
+        for key, value in kwds.items():
+            self[key] = value
+
+    def setdefault(self, key, default=None):
+        'D.setdefault(k[,d]) -> D.get(k,d), also set D[k]=d if k not in D'
+        key = _normalized_headers[key]
+        rv = self._dict.get(key, __marker)
+        if rv is __marker:
+            rv = self._dict.setdefault(key, default)
+            if rv is default:
+                self._as_list[key] = [default]
+        return rv
+
+    def get(self, key, default=None):
+        return self._dict.get(_normalized_headers[key], default)
+
+    def __contains__(self, key):
+        return key in self._dict
+
+    def iterkeys(self):
+        'D.iterkeys() -> an iterator over the keys of D'
+        if PY3:
+            return self._dict.keys()
+        else:
+            return self._dict.iterkeys()
+
+    def itervalues(self):
+        'D.itervalues() -> an iterator over the values of D'
+        if PY3:
+            return self._dict.values()
+        else:
+            return self._dict.itervalues()
+
+    def iteritems(self):
+        'D.iteritems() -> an iterator over the (key, value) items of D'
+        if PY3:
+            return self._dict.items()
+        else:
+            return self._dict.iteritems()
+
+    def keys(self):
+        "D.keys() -> list of D's keys"
+        return self._dict.keys()
+
+    def items(self):
+        "D.items() -> list of D's (key, value) pairs, as 2-tuples"
+        return self._dict.items()
+
+    def values(self):
+        "D.values() -> list of D's values"
+        return self._dict.values()
+
+    __hash__ = None
+
+    def __eq__(self, other):
+        if not isinstance(other, collections.Mapping):
+            return NotImplemented
+        return dict(self.items()) == dict(other.items())
+
+    def __ne__(self, other):
```

Comment:
Is this not the default behavior for ne?

Reply:
Seems not